tests/kernel/spinlock/src/utils.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+
+#define EFLAGS_IF       BIT(9)
+
+/* Macro to save rflags register */
+#define CPU_RFLAGS_SAVE(rflags_ptr)			\
+{							\
+	__asm__ volatile (" pushf");			\
+	__asm__ volatile (" pop %0"			\
+			: "=r" (*(rflags_ptr))		\
+			: /* No inputs */);		\
+}
+
+/* This helper function use to check if interrput enable or not */
+bool check_interrupt_enable_or_not(void)
+{
+#if defined(CONFIG_X86)
```

Comment:
why is there a special case for x86?

can we try to find a way to test this in a arch-neutral way?

Reply:
I put CONFIG_X86 here because function "irq_is_enable()" cannot be used by X86 platform. My goal is to validate when k_spin_lock() called, the interrupt should be disabled. So I try to simply read EFLAG.IF to check if interrupt is disabled or not on X86.  
Maybe the other way is try to implement the irq_is_enable() for x86 platform, then we can keep our testing code here arch-independent. But I am not sure should I do this that way, it seems like making thing more complicated.

Reply:
I try to use another way that Andy recommended to check if interrupt is global masked. And don't treat x86 as a special case.